controllers/tasks.js

```diff
@@ -0,0 +1,94 @@
+const Task = require('../models/Task');
+
+const addTask = (req, res) => {
+  res.render('pages/addTask');
+};
+
+const createTask = async (req, res) => {
+  try {
+    if (req.body.complete) {
+      req.body.completed = true;
+    }
+    await Task.create(req.body);
+    req.session.pendingMessage = 'The task was created.';
+    res.redirect('/tasks');
+  } catch (err) {
+    if (err.name === 'ValidationError') {
+      res.locals.message = Object.values(err.errors)
+        .map((item) => item.message)
+        .join(', ');
+    } else {
+      res.locals.message = 'Something went wrong.';
+    }
+    res.render('pages/addTask');
+  }
+};
+
+const deleteTask = async (req, res) => {
+  try {
+    await Task.findByIdAndDelete(req.params.id);
+    req.session.pendingMessage = 'The task was deleted.';
+    res.redirect('/tasks');
+  } catch (err) {
+    req.session.pendingMessage = 'Something went wrong.';
+    res.redirect('/tasks');
+  }
+};
+
+const editTask = async (req, res) => {
+  try {
+    const task = await Task.findById(req.params.id);
+    res.render('pages/editTask', { task });
+  } catch (err) {
+    req.session.pendingMessage = 'Something went wrong.';
+    res.redirect('/tasks');
+  }
+};
+
+const updateTask = async (req, res) => {
+  let task = false;
+  try {
+    if (req.body.complete) {
+      req.body.completed = true;
+    }
```

Comment:
This isn’t your fault, we're working on getting it updated in the instructions soon, but the `updateTask` is not working if I try to uncheck a task and save. It remains showing as “complete: true”. This is because when an HTML checkbox input is unchecked, its value will not be sent in the form submit message. This means that `req.body.complete` will be undefined. In that case, we need to set the value `completed` to `false` before passing it to the `Task.findByAndUpdate` method here:

```suggestion
    if (req.body.complete) {
      req.body.completed = true;
    } else {
      req.body.completed = false;
    }
```

Reply:
example showing the issue and the Network tab showing that the `complete` field is not passed along in that case

https://github.com/hammaazarok/ejs-demo/assets/9094098/d3e251ad-48f3-4d6e-90f0-f04729dab970